### ISSUE TITLE
Lexical subroutine doesn't return correct name

### ISSUE THREAD

**Issue by nanto** [Perl-Critic/PPI#260]
`PPI` can parse lexical subroutines but its name is incorrect. Run the following code:

```perl
use PPI;
use feature qw(say);

my $doc = PPI::Document->new(\'my sub foo {}');
say $doc->schild(0)->name;
```

Actual output: `sub`

Expected output: `foo`

**Comment by nanto** [comment]
This problem causes Perl-Critic's problem Perl-Critic/Perl-Critic#955.

**Comment by oalders** [comment]
Fixed via #261
